packages/truffle-db/src/workspace/pouch.ts

```diff
@@ -90,68 +87,24 @@ export class Workspace {
     await this.ready;
 
     const { contracts } = input;
-    
+
     return {
       contracts: Promise.all(contracts.map(
         async (contractInput) => {
-          const { name, source } = contractInput;
-          const id = name !== undefined ? soliditySha3(name, source.id) : soliditySha3(source.id);
+          const { name, abi, compilation, sourceContract, constructor: contractConstructor } = contractInput;
+          const id = soliditySha3(jsonStableStringify(name, abi, sourceContract, compilation));
```

Comment:
(Also noted this in the tests, but copying here in case it's easy to miss below)

I think we should wrap the IDs in objects. Here that'd be:
```suggestion
          const id = soliditySha3(jsonStableStringify({ name, abi, sourceContract, compilation }));
```

Reply:
Is this stylistic, or is there a functional reason to do this? Fine with it, just want to be sure I understand the reasoning!